### ISSUE TITLE
front cover image streched

### ISSUE THREAD

**Issue by amejiarosario** [asciidoctor/asciidoctor-pdf#1009]
I have a front page in png but it looked streched no matter what dimensions I gave them.

```
:front-cover-image: image:cover.png[Front Cover,1050,1600]
```

Also, I tried the front cover on PDF 
```
:front-cover-image: image:front-cover.pdf[]
```
and I get the following error
```
 7: from /Users/admejiar/.asdf/installs/ruby/2.5.3/lib/ruby/gems/2.5.0/gems/asciidoctor-pdf-1.5.0.alpha.16/lib/asciidoctor-pdf/converter.rb:184:in `convert_document'
         6: from /Users/admejiar/.asdf/installs/ruby/2.5.3/lib/ruby/gems/2.5.0/gems/asciidoctor-pdf-1.5.0.alpha.16/lib/asciidoctor-pdf/converter.rb:2198:in `layout_cover_page'
         5: from /Users/admejiar/.asdf/installs/ruby/2.5.3/lib/ruby/gems/2.5.0/gems/asciidoctor-pdf-1.5.0.alpha.16/lib/asciidoctor-pdf/prawn_ext/extensions.rb:696:in `import_page'
         4: from /Users/admejiar/.asdf/installs/ruby/2.5.3/lib/ruby/gems/2.5.0/gems/asciidoctor-pdf-1.5.0.alpha.16/lib/asciidoctor-pdf/prawn_ext/extensions.rb:749:in `start_new_page_discretely'
         3: from /Users/admejiar/.asdf/installs/ruby/2.5.3/lib/ruby/gems/2.5.0/gems/asciidoctor-pdf-1.5.0.alpha.16/lib/asciidoctor-pdf/prawn_ext/extensions.rb:731:in `perform_discretely'
         2: from /Users/admejiar/.asdf/installs/ruby/2.5.3/lib/ruby/gems/2.5.0/gems/asciidoctor-pdf-1.5.0.alpha.16/lib/asciidoctor-pdf/prawn_ext/extensions.rb:751:in `block in start_new_page_discretely'
         1: from /Users/admejiar/.asdf/installs/ruby/2.5.3/lib/ruby/gems/2.5.0/gems/prawn-templates-0.1.1/lib/prawn/templates.rb:62:in `start_new_page'
/Users/admejiar/.asdf/installs/ruby/2.5.3/lib/ruby/gems/2.5.0/gems/prawn-2.2.2/lib/prawn/document.rb:694:in `generate_margin_box': undefined method `[]' for nil:NilClass (NoMethodError)
make: *** [pdf] Error 1
```

Not sure, how to make it work with the PDF format or why the dimensions are not working for images and they get streched

**Comment by mojavelinux** [comment]
The cover image is always stretched to the size of the page (which is A4 by default). Therefore, you have to size your image to fit that aspect ratio. The dimensions are currently ignored.

As to why the PDF doesn't work, it's because not all PDFs can be read by the PDF library. It seems the PDF you are trying to use has an incompatibility. I would recommend running the image through a PDF processor (such as ghostscript) to make sure it is clean.

**Comment by mojavelinux** [comment]
It's now possible to control the size of the front cover image. You probably want one of these two options:

```
:front-cover-image: image:front-cover.pdf[fit=contain]
```

(which is the default if no other attributes are specified)

or

```
:front-cover-image: image:front-cover.pdf[fit=cover]
```

The fit attribute always takes priority over the explicit width.

This is available since 1.5.0.beta.1.

**Comment by mojavelinux** [comment]
See #1136 for details about how this got fixed.
